solidity/contracts/AugurConstantProductMarket.sol

```diff
@@ -12,186 +12,107 @@ import { AddressToString } from "./AddressToString.sol";
 contract AugurConstantProduct is ERC20 {
 	using AddressToString for address;
 
-
-	IERC20 public dai = ERC20(Constants.DAI_ADDRESS);
 	IShareToken public shareToken;
 	address public augurMarketAddress;
 	IAugur public constant augur = IAugur(Constants.AUGUR_ADDRESS);
-	uint256 public numTicks;
 	uint256 public INVALID;
 	uint256 public NO;
 	uint256 public YES;
-	uint256 public constant afterFeeDenominator = 1000;
-	uint256 public constant afterFeeNumerator = 950; // 5% fee
+	uint256 public noBalance;
+	uint256 public yesBalance;
+	uint256 public constant feeScope = 1000;
+	uint256 public constant fee = 50; // 5% fee
+
+	uint private unlocked = 1;
+	modifier lock() {
```

Comment:
what causes the lock?

Reply:
Discussed in Discord: It's to guard against reentrancy